### PR TITLE
feat: parse new `banditReferences` field

### PR DESCRIPTION
**Pull request by typotter**
Related BE [PR](https://github.com/Eppo-exp/eppo/pull/9972), [Test Data PR](https://github.com/Eppo-exp/sdk-test-data/pull/48)
🎟️ Ticket: [FF-2713 - Add when bandits have updated to UFC response](https://linear.app/eppo/issue/FF-2713/add-when-bandits-have-updated-to-ufc-response)
🗨️ Slack: ["...We don't need to have a separate polling interval for bandits..."](https://eppo-group.slack.com/archives/C06N7AA6M5F/p1720726503028979?thread_ts=1720671630.501729&cid=C06N7AA6M5F)

### Motivation and Context
`UniversalFlagConfig.Bandits` is now deprecated in favour of the new, more robust field, `banditReferences`. The new schema is extensible; the initial intent was to support referencing specific bandit model versions in the UFC response in order to optimize bandit loading.

### Description
This PR migrates the codebase to this new field of the UFCResponse. In addition to reading the `banditReferences` field, there is considerable renaming (s/variation/reference) in class and method names to accommodate the change in structure.
This pull also takes advantage of the refactor to move the `BanditFlagVariation` class out of the DTO\Bandit namespace as that space is for DTOs related more to the bandit model response.

### Testing
This PR does not introduce any new functionality; existing testing passes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,7 +30,7 @@ help: Makefile
 testDataDir := tests/data/
 tempDir := ${testDataDir}temp/
 gitDataDir := ${tempDir}sdk-test-data/
-branchName := main
+branchName := tp/banditRefs
```
Comment (typotter) [inline Makefile:33]: to be reverted when sdk-test-data PR is merged

File: src/Bandits/BanditReferenceIndexer.php [hunk 2]
```diff
@@ -2,20 +2,23 @@
...
-class BanditVariationIndexer implements IBanditVariationIndexer
+class BanditReferenceIndexer implements IBanditReferenceIndexer
 {
-    // By just serializing the indexed variations, we cut down on cache size.
+    // magic methods to store just the underlying data when serialized to cache.
     public function __serialize(): array
     {
-        return $this->banditFlags;
+        return [
+            'flagIndex' => $this->flagIndex
```
Comment (typotter) [inline src/Bandits/BanditReferenceIndexer.php:15]: future work will have multiple private fields stored when serialized. Using a map here prepares for this.

File: src/Bandits/BanditReferenceIndexer.php [hunk 3]
```diff
@@ -70,31 +73,36 @@ private function setVariations(array $banditVariations): void
      */
     public function getBanditByVariation(string $flagKey, string $variation): ?string
     {
-        return $this->banditFlags[$flagKey][$variation] ?? null;
+        return $this->flagIndex[$flagKey][$variation] ?? null;
     }
 
-    public function isBanditFlag(string $flagKey): bool
```
Comment (typotter) [inline src/Bandits/BanditReferenceIndexer.php:76]: This method no longer needed

File: src/Bandits/IBanditReferenceIndexer.php [hunk 4]
```diff
@@ -16,14 +16,6 @@ interface IBanditVariationIndexer
      */
     public function getBanditByVariation(string $flagKey, string $variation): ?string;
 
-    /**
```
Comment (typotter) [inline src/Bandits/IBanditReferenceIndexer.php:19]: method not needed

File: tests/Bandits/BanditReferenceIndexerTest.php [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+final class BanditReferenceIndexerTest extends TestCase
+{
+    /**
+     * @var array<string, BanditReference>
+     */
+    private static array $banditReferences = [];
+
+    public static function setUpBeforeClass(): void
+    {
+        // Bandits:Flags can be m:n
```
Comment (giorgiomartini0) [inline tests/Bandits/BanditReferenceIndexerTest.php:20]: Good future-proofing (test)!

**Review by giorgiomartini0 — APPROVED** [review]

**Pull request merged**